tests/ui/unnecessary_unwrap.stderr

```diff
@@ -0,0 +1,36 @@
+error: you checked before that `unwrap()` cannot fail, instead of checking and unwrapping, it's better to use `if let` or `match`
```

Comment:
Do you mind adjusting the output for this lint while you're here? I find it hard to read currently. Something like this would be nice:

```text
error: called `unwrap` on `opt` after checking its variant with `is_some`
  --> $DIR/unnecessary_unwrap.rs:11:17
   |
LL |     if opt.is_some() {
   |        ------------- the variant is checked here
LL |         let _ = opt.expect("an error message");
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `expect` is called here
help: try using `if let` or `match` instead
```
An actual code suggestion would be even better, but not necessary.

Reply:
I think that means I need something like 

```rust
&format!(
    "called `{}` on `{}` after checking its variant with `{}`",
    method_name.ident.name,
    unwrappable.ident,
    unwrappable.check,
),
```

However, `Path` and `Expr` do not implement `Display`, and I don't see anything obvious to get what I want. Any clues on where to look for examples?

Reply:
Alright, a revised version is up.

> An actual code suggestion would be even better

Any tips on where to look for how to do that?

Reply:
You would need to add `diag.span_suggestion(span, suggestion)` to the closure in `span_lint_and_then`. The `span` would be from the `Expr` of `opt.is_some()` and the suggestion would be like `"let Some(value) = opt"`.